pkg/apis/openebs.io/upgrade/v1alpha2/types.go

```diff
@@ -0,0 +1,171 @@
+	CStorVolume      *CStorVolume      `json:"cstorVolume,omitempty" protobuf:"bytes,2,opt,name=cstorVolume"`
+	CStorPool        *CStorPool        `json:"cstorPool,omitempty" protobuf:"bytes,3,opt,name=cstorPool"`
+	StoragePoolClaim *StoragePoolClaim `json:"storagePoolClaim,omitempty" protobuf:"bytes,3,opt,name=storagePoolClaim"`
+}
+
+// JivaVolume is the ResourceType for jiva volume
+type JivaVolume struct {
+	PVName string         `json:"pvName,omitempty" protobuf:"bytes,1,name=pvName"`
+	Flags  *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// CStorVolume is the ResourceType for cstor volume
+type CStorVolume struct {
+	PVName string         `json:"pvName,omitempty" protobuf:"bytes,1,name=pvName"`
+	Flags  *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// CStorPool is the ResourceType for cstor pool
+type CStorPool struct {
+	PoolName string         `json:"poolName,omitempty" protobuf:"bytes,1,name=poolName"`
+	Flags    *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// StoragePoolClaim is the ResourceType for storage pool claim
+type StoragePoolClaim struct {
+	PoolName string         `json:"poolName,omitempty" protobuf:"bytes,1,name=poolName"`
+	Flags    *ResourceFlags `json:"flags,omitempty" protobuf:"bytes,2,opt,name=flags"`
+}
+
+// ResourceFlags provides additional options for a particular resource
+type ResourceFlags struct {
+	IgnoreStepsOnError []string `json:"ignoreStepsOnError,omitempty" protobuf:"bytes,1,opt,name=ignoreStepsOnError"`
+}
+
+// UpgradeTaskStatus provides status of a cas volume
+type UpgradeTaskStatus struct {
+	// Phase indicates if a volume is available, pending or failed
+	Phase                   UpgradePhase              `json:"phase,omitempty" protobuf:"bytes,1,opt,name=phase,casttype=UpgradePhase"`
+	StartTime               metav1.Time               `json:"startTime,omitempty" protobuf:"bytes,2,opt,name=startTime"`
+	CompletedTime           metav1.Time               `json:"completedTime,omitempty" protobuf:"bytes,3,opt,name=completedTime"`
+	UpgradeDetailedStatuses []UpgradeDetailedStatuses `json:"upgradeDetailedStatuses,omitempty" protobuf:"bytes,4,rep,name=upgradeDetailedStatuses"`
+}
+
+// UpgradeDetailedStatuses represents the latest available observations
+// of a UpgradeTask current state.
+type UpgradeDetailedStatuses struct {
+	Step          UpgradeStep `json:"step,omitempty" protobuf:"bytes,1,opt,name=step"`
+	StartTime     metav1.Time `json:"startTime,omitempty" protobuf:"bytes,2,opt,name=startTime"`
+	LastUpdatedAt metav1.Time `json:"lastUpdatedAt,omitempty" protobuf:"bytes,3,opt,name=lastUpdatedAt"`
+	State         State       `json:"state" protobuf:"bytes,4,opt,name=state"`
+}
+
+// State represents the state of the step performed during the upgrade.
+type State struct {
+	Phase StatePhase `json:"phase,omitempty" protobuf:"bytes,1,opt,name=phase"`
```

Comment:
Similar to above - `Phase` to `Status` and `StatePhase` to `StatusTypes`

Reply:
Done